### DISCUSSION TITLE
Cannot set exception-handling-method to "js"

### DISCUSSION THREAD

**Discussion by tritao** [bfgroup/b2#337] [Q&A]
`emscripten.jam` has the following:

```
toolset.flags emscripten.compile.c++ OPTIONS <exception-handling>on/<exception-handling-method>js : -fexceptions ;
toolset.flags emscripten.link        OPTIONS <exception-handling>on/<exception-handling-method>js : -fexceptions ;
toolset.flags emscripten.compile.c++ OPTIONS <exception-handling>on/<exception-handling-method> : -fwasm-exceptions ;
toolset.flags emscripten.link        OPTIONS <exception-handling>on/<exception-handling-method> : -fwasm-exceptions ;
```

I am trying to get boost to compile with `-fexceptions` flag, right now it is defaulting to `-fwasm-exceptions`.

I have tried to call b2 with `b2 exception-handling-method=js ...`.

I have also tried to add `option.set exception-handling-method : js ;` to `project-config.jam`.

Neither option seems to work, only way I have gotten a workaround is to hardcode `-fwasm-exceptions` to `-fexceptions` in `emscripten.jam`.

Does anyone know the right way to fix this?

**Comment by grisumbras** [comment]
Does `b2 exception-handling-method=js` generate any errors or does it simply not work?
